iothub/device/src/PayloadCollection.cs

```diff
@@ -20,7 +20,7 @@ public abstract class PayloadCollection : IEnumerable<KeyValuePair<string, objec
         /// <summary>
         /// The underlying collection for the payload.
         /// </summary>
-        public IDictionary<string, object> Collection { get; private set; } = new Dictionary<string, object>();
+        public IDictionary<string, object> Collection { get; } = new Dictionary<string, object>();
```

Comment:
Why do we expose the collection when the containing class `is` an IEnumerable<KeyValuePair>string, object>>?

Reply:
I should be able to make this collection internal. I don't see a need for users to be accessing this explicitly.